### ISSUE TITLE
Use of `--no-1mm-upfront` crashes sometimes

### ISSUE THREAD

**Issue by BenLangmead** [BenLangmead/bowtie2#250]
```
$ ./bowtie2-align-s -x example/index/lambda_virus -U example/reads/reads_1.fq --no-1mm-upfront -u 1
Segmentation fault: 11
```

When I load it in gdb, it puts the crash somewhere in the destructor for `APtrWrap`.  If I remove --no-1mm-upfront, the issue goes away.  It also seems to go away when I use the `-debug` binary.

**Comment by BenLangmead** [comment]
This happens on my Mac.  Versions:

```
$ c++ --version
Apple LLVM version 10.0.1 (clang-1001.0.46.3)
Target: x86_64-apple-darwin18.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
$ uname -a
Darwin ben-mbp-2017.lan 18.5.0 Darwin Kernel Version 18.5.0: Mon Mar 11 20:40:32 PDT 2019; root:xnu-4903.251.3~3/RELEASE_X86_64 x86_64 i386 MacBookPro14,3 Darwin
```

**Comment by ch4rr0** [comment]
This change is available in v2.4.0 and later.
